source/components/localisation/utils.ts

```diff
@@ -7,37 +7,93 @@
 */
 
 import { useSelector } from 'react-redux'
+import { AppState } from '../../reducers/rootReducer'
 import frenchCountryPrepositions from './frenchCountryPrepositions.yaml'
 
-export const defaultModel = 'FR'
+export type RegionCode = string
 
-export const supportedRegion = (inputCode) => {
-	const supportedRegions = useSelector((state) => state.supportedRegions)
-	return supportedRegions[inputCode]
+export type RegionAuthor = {
+	nom: string
+	url?: string
 }
 
-export const getFlag = (inputCode) => {
-	const regionParams = supportedRegion(inputCode)
-	const code = regionParams?.fr?.drapeau ?? inputCode
-	return getFlagImgSrc(code)
+export type RegionParams = {
+	code: RegionCode
+	nom: string
+	gentilé: string
+	authors?: RegionAuthor[]
+	drapeau?: string
+}
+
+export type Region = {
+	fr: RegionParams
+	en: RegionParams
+}
+
+export type SupportedRegions = {
+	[code: RegionCode]: Region
 }
 
-export const getModelFlag = (inputCode) => {
-	const regionParams = supportedRegion(inputCode)
-	const code = regionParams ? regionParams?.drapeau ?? inputCode : defaultModel
+export type Localisation = {
+	country: { code: RegionCode; name: string }
+	userChosen: boolean
+}
+
+export const defaultModelRegionCode = 'FR'
+
+/**
+ * This function is not pure, it uses useSelector, dont't call it
+ * conditionally !!
+ */
+export function getSupportedRegion(
```

Comment:
Toutes les fonctions qui utilisent des hooks sont des hooks et devraient être nommés avec le préfix `use`, par exemple `useGetSupportedRegionCode`. Je me serais clairement fait avoir ici moi aussi, IT'S A TRAP

Reply:
Je savais pas je vais modifier ça :+1: